clouddriver-google-common/src/main/groovy/com/netflix/spinnaker/clouddriver/googlecommon/deploy/GoogleCommonSafeRetry.groovy

```diff
@@ -38,50 +46,93 @@ abstract class GoogleCommonSafeRetry {
+                                            Long maxWaitInterval,
+                                            Long retryIntervalBase,
+                                            Long jitterMultiplier,
+                                            Long maxRetries) {
+    SafeRetryState state = new SafeRetryState()
     try {
       task?.updateStatus phase, "Attempting $action of $resource..."
-      return operation()
+      state.finalResult = operation()
+      state.success = true
+      return state
     } catch (GoogleJsonResponseException | SocketTimeoutException | SocketException _) {
       log.warn "Initial $action of $resource failed, retrying..."
 
-      int tries = 1
-      Exception lastSeenException = null
-      while (tries < maxRetries) {
+      while (state.tries < maxRetries) {
         try {
-          tries++
+          def tries = ++state.tries
           // Sleep with exponential backoff based on the number of retries. Add retry jitter with Math.random() to
           // prevent clients syncing up and bursting at regular intervals. Don't wait longer than a minute.
           Long thisIntervalWait = TimeUnit.SECONDS.toMillis(Math.pow(retryIntervalBase, tries) as Integer)
           sleep(Math.min(thisIntervalWait, maxWaitInterval) + Math.round(Math.random() * jitterMultiplier))
           log.warn "$action $resource attempt #$tries..."
-          return operation()
+          state.finalResult = operation()
+          state.success = true
+          return state
         } catch (GoogleJsonResponseException jsonException) {
           if (jsonException.statusCode in successfulErrorCodes) {
             log.warn "Retry $action of $resource encountered ${jsonException.statusCode}, treating as success..."
-            return null
+            state.success = true
+            return state
           } else if (jsonException.statusCode in retryCodes) {
             log.warn "Retry $action of $resource encountered ${jsonException.statusCode} with error message: ${jsonException.message}. Trying again..."
           } else {
             throw jsonException
           }
-          lastSeenException = jsonException
+          state.lastSeenException = jsonException
         } catch (SocketTimeoutException toEx) {
           log.warn "Retry $action timed out again, trying again..."
-          lastSeenException = toEx
+          state.lastSeenException = toEx
         }
       }
+    }
+    return state
+  }
 
-      if (lastSeenException && lastSeenException instanceof GoogleJsonResponseException) {
+  protected Object determineFinalResult(SafeRetryState state,
+                                        String action,
+                                        String resource) {
+    if (state.success) {
+      return state.finalResult
+    }
+    Exception lastSeenException = state.lastSeenException
+    int tries = state.tries
+    if (lastSeenException && lastSeenException instanceof GoogleJsonResponseException) {
```

Comment:
You can drop the `lastSeenException &&`. `instanceof` returns `false` when operating on a `null`.